### PR TITLE
feat: insert `use client` automatically at compilations instead  of  hardcode

### PR DESCRIPTION
**Pull request by Dunqing**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

Related PR: https://github.com/ant-design/antd-tools/pull/273

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     insert `"use client"` automatically at compilations       |
| 🇨🇳 Chinese |   编译时自动插入 `"use client"`        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 5b09936</samp>

Removed the `'use client';` directive from various TypeScript files in the `components` folder, as it is not a valid or necessary directive for TypeScript or browser compatibility.

### <samp>🤖 Generated by Copilot at 5b09936</samp>

*  Removed invalid `'use client';` directive from 20 TypeScript files ([link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-18949a009c0a2f7f47a0136b5808eade7a157b3d78661edc0f75463594e2ad5dL1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-ea17437115f696bb5e9eda1c71130b498ea13d5aee96f5d40bce255e8c68b8e7L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-bce8b84c85cec197fd50b0e18482c4c8d0d8a05eb51d9714b869f4d9f0abe6caL1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-95b0dcfbdd9a195d1d0141272a065d901452a60a60632bd0e30f74e80eb85a6bL1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-c55203058d3a4064d8f52a5b4e7d7770153d215966d276d6814011d99268f76dL1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-eafdeaa24b5e2771367e4ea1383becc626c63072ccd786648d000fa1df282fcaL1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-22587f2ed95b1c480472e426d68bdb6cc32d38b3134e9aa5a32101b973312618L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-da5ca906e2a2a35f0eeeacfcbd4cf2bc6288817ddd292e5ce858f150623606d6L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-e17ebc19f2ad32b347a03e08a1b4f367a2280027a3e42d2ee4e4c737a57a2243L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-013735da2a9e27cd090e2a7b89864881fcab5e33ff74b538079f42b82ccd3d24L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-8a8c225c4369c726cc4f6f3c5dbe979d57e69d5d76a3a0d04fe56016d5aa2c54L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-1158da0914c171fa74b4fa496bef7272ca0776317fb5e222bebdf11477b191fbL1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-8220e5af92084a379caeaf225cdad0f0090e0858855f703909a9376e2c037026L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-6ffd57ad327ed714187e4e46d7289dfe4b84c6e51b08535c56115f3b2847143bL1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-2d2d4fb9566406cae122a321d0c5bb8e9006d00b7a7804bc408ba0ad5f81c098L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-937e920c7edf3c1b09cde8ed592d2ccc37d8f6eaee8074582eec32dd70c90850L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56L1-L2), [link](https://github.com/ant-design/ant-design/pull/44608/files?diff=unified&w=0#diff-78eb7aa1662e48dc14665cbcead280795a60503742024dd031b87311186be5eeL1-L2))

### CODE REVIEW

**Review by kiner-tang — APPROVED** [review]
LGTM

**Comment by MadCcc** [comment]
Checklist:
- [ ] lib 下组件 index 和源文件（原 tsx）
- [ ] es 下组件 index 和源文件（原 tsx）
- [ ] dist 不包含 'use client'

**Comment by afc163** [comment]
Checklist:

    lib 下组件 index 和源文件（原 tsx）
    es 下组件 index 和源文件（原 tsx）
    dist 不包含 'use client'


得有用例。

**Comment by MadCcc** [comment]
## size-limit report 📦 
| Path                              | Size      |
| --------------------------------- | --------- |
| `./dist/antd.min.js`              | 387.73 KB |
| `./dist/antd-with-locales.min.js` | 446.79 KB |

**Comment by MadCcc** [comment]
> 得有用例。

@Dunqing 可以用 dekko 简单测一下

File: tests/dekko/use-client.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const $ = require('dekko');
+const chalk = require('chalk');
+const fs = require('fs');
+
+const includeUseClient = (filename) =>
+  fs.readFileSync(filename).toString().includes('"use client"');
```
Comment (afc163) [inline tests/dekko/use-client.test.js:6]:
```suggestion
  fs.readFileSync(filename).toString().startsWith('"use client"');
```

Comment (afc163) [inline tests/dekko/use-client.test.js:6]:
```suggestion
const startsWithUseClient = (filename) =>
  fs.readFileSync(filename).toString().startsWith('"use client"');
```

Comment (Dunqing) [reply tests/dekko/use-client.test.js:6]: lib 下不对
image

**Review by MadCcc — APPROVED** [review]

**Pull request merged**